hikari/events/channel_events.py

```diff
@@ -779,3 +787,204 @@ async def fetch_guild_webhooks(self) -> typing.Sequence[webhooks.PartialWebhook]
+class OwnThreadMemberUpdateEvent(GuildThreadEvent):
+    """Event fired when the bot's own member is updated in a thread."""
+
+    app: traits.RESTAware = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    member: channels.ThreadMember = attr.field()
+    """The member that was updated."""
+
+    guild_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    @property
+    def thread_id(self) -> snowflakes.Snowflake:
+        # <<inherited docstring from GuildThreadEvent>>.
+        return self.member.thread_id
+
+
+@base_events.requires_intents(intents.Intents.GUILDS, intents.Intents.GUILD_MEMBERS)
+@attr_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class ThreadMembersUpdateEvent(GuildThreadEvent):
+    """Event fired when other thread members are updated."""
+
+    app: traits.RESTAware = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    thread_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    guild_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    approximate_member_count: int = attr.field(eq=False, hash=False, repr=True)
+    """Approximate count of members in the thread channel.
+
+    !!! warning
+        This stops counting at 50.
+    """
+
+    added_members: typing.Mapping[snowflakes.Snowflake, channels.ThreadMember] = attr.field()
+    removed_member_ids: typing.Sequence[snowflakes.Snowflake] = attr.field()
+    guild_members: typing.Mapping[snowflakes.Snowflake, guilds.Member] = attr.field()
+    guild_presences: typing.Mapping[snowflakes.Snowflake, presences.MemberPresence] = attr.field()
+
+
+@base_events.requires_intents(intents.Intents.GUILDS)
+@attr_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class ThreadListSyncEvent(shard_events.ShardEvent):
+    """Event fired to sync threads when the bot gains access to one or more channels."""
+
+    app: traits.RESTAware = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    guild_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    channel_ids: typing.Optional[typing.Sequence[snowflakes.Snowflake]] = attr.field()
+
+    threads: typing.Mapping[snowflakes.Snowflake, channels.GuildThreadChannel] = attr.field()
```

Comment:
Missing docstrings

Reply:
Yeah I haven't done docs yet

Reply:
Yea, all good. Just wanted to point them out :)